Move js-specific URL parser tests into their own file

simonwuelker opened the pull request:
The test in question contains UTF-16 surrogates, but the input to a url parser is defined to be a "scalar value string", which is a string containing *no* surrogates.

Fixes https://github.com/web-platform-tests/wpt/issues/46941, refer to this issue for more information.

annevk commented:
I think you are correct that if you feed this inputs directly into the parser it would be wrong, but we're feeding these inputs into `new URL()` which takes `USVString` which handles the conversion.

I think it's fair to say we should decouple that from this JSON file so it's useful more generally, but I think that would require adding this as a standalone test somehow that browsers and other JS clients (thinking of Deno and Node.js) would still run as it's very meaningful to them.

domenic commented:
Concrete proposal: break urltestdata.json up into two files, maybe urltestdata.json and urltestdata-javascript-only.json. The latter would get a comment at the top explaining the very rare cases in which it should be used. And then modify all the tests that consume urltestdata.json to consume both files.

simonwuelker committed:
Split off javascript-only URL parser tests

Previously, urltestdata.json contained a test
for behaviour that is specific to javascripts
"new URL()". Tests like this should instead live
in their own test file, so implementations that
are not concerned with javascript don't have
to bother with them.

simonwuelker committed:
Use both regular and js-only url parser tests everywhere

annevk commented on url/README.md:
```diff
@@ -1,11 +1,13 @@
+## urltestdata.json / urltestdata-javascript-only.json
 
 `resources/urltestdata.json` contains URL parsing tests suitable for any URL parser implementation.
+`resources/urltestdata-javascript-only.json` contains URL parsing tests specifically meant 
+for javascripts `new URL()` function.
```
```suggestion
`resources/urltestdata-javascript-only.json` contains URL parsing tests specifically meant 
for JavaScript's `URL()` class as well as other languages accepting non-scalar-value strings.
```

annevk commented on url/resources/urltestdata-javascript-only.json:
```diff
@@ -0,0 +1,18 @@
+        "search": "?%EF%BF%BD%F0%90%9F%BE%EF%BF%BD%EF%B7%90%EF%B7%8F%EF%B7%AF%EF%B7%B0%EF%BF%BE%EF%BF%BF",
+        "hash": ""
+      }
+]
```
Please add a newline.

annevk reviewed (commented):
Thanks, couple nits from my side, but with that this seems reasonable.

@lucacasonato @jasnell any concerns from your side?

annevk commented:
Also seems you have some trailing whitespace to remove as per the lint bot.
